lib/_http_incoming.js

```diff
@@ -121,7 +121,7 @@ IncomingMessage.prototype._addHeaderLines = function(headers, n) {
 // and drop the second. Extended header fields (those beginning with 'x-') are
 // always joined.
 IncomingMessage.prototype._addHeaderLine = function(field, value, dest) {
-  field = field.toLowerCase();
+  field = field.toLowerCase().trim();
```

Comment:
Unfortunately we can't. Consider, `Content-Type` for instance. If I move the `trim()` into the `default:` clause then send the following request:

```
POST / HTTP/1.1
Content-Type: text/plain
Content-Type    : text/foo
```

The headers that end up reported on the server side are:

```js
{
  'content-type': ['text/plain', 'text/foo']
}
```

However, if I send the second content type without the whitespace, it comes to:
```
POST / HTTP/1.1
Content-Type: text/plain
Content-Type: text/foo
```

```js
{ 'content-type': 'text/plain' }
```

One thing I *could* do is a quick `charAt` check on the field name is see if it's `\s` or `\t` and trim only if it is.

Modifying the http-parser to do the trimming turns out to be quite a bit more involved. I'll keep investigating that change. @indutny may have some ideas.